src/bigquery.ts

```diff
@@ -729,6 +732,60 @@ export class BigQuery extends common.Service {
     return BigQuery.geography(value);
   }
 
+  /**
+   * Return a value's provided type.
+   *
+   * @private
+   *
+   * @throws {error} If the type provided is invalid.
+   *
+   * @see [Data Type]{@link https://cloud.google.com/bigquery/data-types}
+   *
+   * @param {*} providedType The type.
+   * @returns {string} The valid type provided.
+   */
+  // tslint:disable-next-line no-any
+  static getProvidedType_(providedType: any): ValueType {
+    const VALID_TYPES = [
+      'DATE',
+      'DATETIME',
+      'TIME',
+      'TIMESTAMP',
+      'BYTES',
+      'NUMERIC',
+      'BOOL',
+      'INT64',
+      'FLOAT64',
+      'STRING',
+      'GEOGRAPHY',
+    ];
+
+    if (is.array(providedType)) {
+      return {
+        type: 'ARRAY',
+        arrayType: BigQuery.getProvidedType_(providedType[0]),
+      };
+    } else if (is.object(providedType)) {
+      return {
+        type: 'STRUCT',
+        structTypes: Object.keys(providedType).map(prop => {
+          return {
+            name: prop,
+            type: BigQuery.getProvidedType_(providedType[prop]),
+          };
+        }),
+      };
+    } else if (is.boolean(providedType)) {
```

Comment:
Is it valid/expected for a user to provide a boolean as a type?

Reply:
Yeah, it's a BigQuery type that can be detected so they should be able to provide it as well.

Reply:
Oh, I see. Fixed.